[release] v7.22.4 (#15655)
mui/mui-x: CHANGELOG.md

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 7.22.4

_Nov 28, 2024_

> We'd like to offer a big thanks to the 8 contributors who made this release possible. Here are some highlights ✨:

Reviewer: ```suggestion
We'd like to offer a big thanks to the 9 contributors who made this release possible. Here are some highlights ✨:
```

- 📚 Documentation improvements

> - 🌍 Improve Dutch, Spanish, Swedish, Portuguese, Chinese locales on the Data Grid and Date Pickers.

Reviewer: Usually we do one line for the grid and one line for the pickers

- 🐞 Bugfixes

Special thanks go out to the community contributors who have helped make this release possible:

> @dloeda, @headironc, @mathzdev, @nphmuller.

Reviewer 2: ```suggestion
@dloeda, @headironc, @mathzdev, @nphmuller, @lhilgert9.
```

Following are all team members who have contributed to this release:

> @oliviertassinari, @arminmeh, @KenanYusuf.

Reviewer: ```suggestion
@oliviertassinari, @arminmeh, @KenanYusuf, @flaviendelangle.
```

> @oliviertassinari, @arminmeh, @KenanYusuf.

Reviewer 2: ```suggestion
@oliviertassinari, @arminmeh, @KenanYusuf, @flaviendelangle, @MBilalShafi.
```

<!--/ HIGHLIGHT_ABOVE_SEPARATOR /-->

### Data Grid

#### `@mui/x-data-grid@v7.22.4`

> - [docs][data grid] Add documentation for the list view feature (#15344) (#15635) @KenanYusuf

Reviewer: Should move to the docs and remove `[data grid]` (we might want to discuss how to correctly prefix the `[docs]` PRs btw, the more product we have, the weirder it is to group the docs PR together instead of moving them to their respective packages IMHO

> - [DataGrid] Change test dom check from /jsdom/ to /jsdom|HappyDOM/. (@jedesroches) (#15642)

Reviewer: ```suggestion
- [DataGrid] Change test dom check from `/jsdom/` to `/jsdom|HappyDOM/`. (@jedesroches) (#15642)
```

> - [DataGrid] Fix last separator not being hidden when grid is scrollable (@KenanYusuf) (#15551)

Reviewer 2: ```suggestion
- [DataGrid] Fix last separator not being hidden when grid is scrollable (#15551) @KenanYusuf
```

> - [DataGrid] Fix row-spanning in combination with column-pinning (@lhilgert9) (#15460)

Reviewer 2: ```suggestion
- [DataGrid] Fix row-spanning in combination with column-pinning (#15460) @lhilgert9
```

> - [DataGrid] Improve resize performance (@lauri865) (#15592)

Reviewer 2: ```suggestion
- [DataGrid] Improve resize performance (#15592) @lauri865
```

> - [DataGrid] Support column virtualization with dynamic row height (@cherniavskii) (#15567)

Reviewer 2: ```suggestion
- [DataGrid] Support column virtualization with dynamic row height (#15567) @cherniavskii
```

- [l10n] Improve Chinese (zh-CN) locale (#15570) @headironc
- [l10n] Improve Portuguese (pt-PT) locale (#15561) @mathzdev

#### `@mui/x-data-grid-pro@v7.22.4` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-data-grid@v7.22.4`, plus:

- [DataGridPro] Fix header filtering with `boolean` column type (@k-rajat19) (#15640) @arminmeh

> - [DataGridPro] Fix pagination state not updating if the data source response has no rows (@zinoroman) (#15643)

Reviewer 2: ```suggestion
- [DataGridPro] Fix pagination state not updating if the data source response has no rows (#15643) @zinoroman
```

> - [DataGridPro] Fix selection propagation issue on initialization (@MBilalShafi) (#15593)

Reviewer 2: ```suggestion
- [DataGridPro] Fix selection propagation issue on initialization (#15593) @MBilalShafi
```

#### `@mui/x-data-grid-premium@v7.22.4` [![premium](https://mui.com/r/x-premium-svg)](https://mui.com/r/x-premium-svg-link 'Premium plan')

Same changes as in `@mui/x-data-grid-pro@v7.22.4`.

### Date and Time Pickers

#### `@mui/x-date-pickers@v7.22.4`

- [l10n] Improve Dutch (nl-NL) locale (#15564) @nphmuller

> - [pickers] Use `props.referenceDate` timezone when `props.value` and `props.defaultValue` are not defined (@flaviendelangle) (#15544)

Reviewer: ```suggestion
- [pickers] Fix DST issue with `America/Asuncion` timezone and `AdapterMoment` #15653 (@flaviendelangle)
- [pickers] Use `props.referenceDate` timezone when `props.value` and `props.defaultValue` are not defined (@flaviendelangle) (#15544)
```

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 7.22.4

_Nov 28, 2024_

We'd like to offer a big thanks to the 10 contributors who made this release possible. Here are some highlights ✨:

> - 📚 Documentation improvements

Reviewer 3: ```suggestion
- ⚛️ React 19 support
- 📚 Documentation improvements
```

Should this make it a minor release?

- 🌍 Improve Spanish, Portuguese, Chinese locales on the Data Grid component.
- 🌍 Improve Dutch locales on the Date and Time Pickers components.
- 🐞 Bugfixes

Special thanks go out to the community contributors who have helped make this release possible:
@dloeda, @headironc, @mathzdev, @nphmuller, @lhilgert9.
Following are all team members who have contributed to this release:
@oliviertassinari, @arminmeh, @KenanYusuf, @flaviendelangle, @MBilalShafi.

<!--/ HIGHLIGHT_ABOVE_SEPARATOR /-->

### Data Grid

#### `@mui/x-data-grid@v7.22.4`

- [docs][data grid] Add documentation for the list view feature (#15344) (#15635) @KenanYusuf

> - [DataGrid] Change test dom check from `/jsdom/` to `/jsdom|HappyDOM/`. (#15642) @jedesroches

Reviewer 3: ```suggestion
- [DataGrid] React 19 support (#15557) @arminmeh
- [DataGrid] Change test dom check from `/jsdom/` to `/jsdom|HappyDOM/`. (#15642) @jedesroches
```

- [DataGrid] Fix last separator not being hidden when grid is scrollable (#15551) @KenanYusuf
- [DataGrid] Fix row-spanning in combination with column-pinning (#15460) @lhilgert9
- [DataGrid] Improve resize performance (#15592) @lauri865
- [DataGrid] Support column virtualization with dynamic row height (#15567) @cherniavskii
- [l10n] Improve Chinese (zh-CN) locale (#15570) @headironc
- [l10n] Improve Portuguese (pt-PT) locale (#15561) @mathzdev

#### `@mui/x-data-grid-pro@v7.22.4` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-data-grid@v7.22.4`, plus:

> - [DataGridPro] Fix header filtering with `boolean` column type (@k-rajat19) (#15640) @arminmeh

Reviewer 3: ```suggestion
- [DataGridPro] Fix header filtering with `boolean` column type (#15640) @k-rajat19
```

- [DataGridPro] Fix pagination state not updating if the data source response has no rows (#15643) @zinoroman
- [DataGridPro] Fix selection propagation issue on initialization (#15593) @MBilalShafi

#### `@mui/x-data-grid-premium@v7.22.4` [![premium](https://mui.com/r/x-premium-svg)](https://mui.com/r/x-premium-svg-link 'Premium plan')

Same changes as in `@mui/x-data-grid-pro@v7.22.4`.

### Date and Time Pickers

#### `@mui/x-date-pickers@v7.22.4`

- [l10n] Improve Dutch (nl-NL) locale (#15564) @nphmuller

> - [pickers] Fix DST issue with `America/Asuncion` timezone and `AdapterMoment` (#15653) @flaviendelangle

Reviewer 3: ```suggestion
- [pickers] React 19 support (#15557) @arminmeh
- [pickers] Fix DST issue with `America/Asuncion` timezone and `AdapterMoment` (#15653) @flaviendelangle
```

- [pickers] Use `props.referenceDate` timezone when `props.value` and `props.defaultValue` are not defined (#15544) @flaviendelangle

#### `@mui/x-date-pickers-pro@v7.22.4` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-date-pickers@v7.22.4`.

### Charts

#### `@mui/x-charts@v7.22.4`

> - [charts] Prevent invalid `releasePointerCapture` (#15609) @alexfauquette

Reviewer 3: ```suggestion
- [charts] React 19 support (#15557) @arminmeh
- [charts] Prevent invalid `releasePointerCapture` (#15609) @alexfauquette
```

#### `@mui/x-charts-pro@v7.22.4` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-charts@v7.22.4`.

### Tree View

#### `@mui/x-tree-view@v7.22.4`

> No changes since `@mui/x-tree-view@v7.22.3`.

Reviewer 3: ```suggestion
- [TreeView] React 19 support (#15557) @arminmeh

#### `@mui/x-tree-view-pro@7.22.4` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-tree-view@7.22.4`.
```

### Docs

- [docs] Add data caching to lazy loaded detail panel demo (#15555) @cherniavskii
- [docs] Remove selectors section from list view docs (#15639) @KenanYusuf
- [docs][data grid] Add documentation for the list view feature (#15344) (#15635) @KenanYusuf

### Core

- [core] Update @mui/monorepo (#15574) @oliviertassinari

> - React 19 support (#15557) @arminmeh

Reviewer 3: ```suggestion
```

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 7.23.0

> _Nov 28, 2024_

Reviewer 3: ```suggestion
_Nov 29, 2024_
```

We'd like to offer a big thanks to the 10 contributors who made this release possible. Here are some highlights ✨:

> 

Reviewer 4: ```suggestion
- ✨ Support displaying the Data Grid component in a single-column with the [List view feature](https://mui.com/x/react-data-grid/list-view/).

  https://github.com/user-attachments/assets/95d8fcc0-fe16-4f08-b46b-19d4d4caa6b9

```

Reviewer 4: Hopefully the video works when we publish the release 🤨

Reviewer 4: I like that a lot 👍

- ⚛️ React 19 support
- 📚 Documentation improvements
- 🌍 Improve Spanish, Portuguese, Chinese locales on the Data Grid component.

> - 🌍 Improve Dutch locales on the Date and Time Pickers components.

Reviewer 5: ```suggestion
- 🌍 Improve Dutch locale on the Date and Time Picker components.
```

- [pickers] Fix DST issue with `America/Asuncion` timezone and `AdapterMoment` (#15653) @flaviendelangle
- [pickers] Use `props.referenceDate` timezone when `props.value` and `props.defaultValue` are not defined (#15544) @flaviendelangle
- [l10n] Improve Dutch (nl-NL) locale (#15564) @nphmuller

#### `@mui/x-date-pickers-pro@v7.23.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-date-pickers@v7.23.0`.

### Charts

#### `@mui/x-charts@v7.23.0`

- [charts] React 19 support (#15557) @arminmeh
- [charts] Prevent invalid `releasePointerCapture` (#15609) @alexfauquette

#### `@mui/x-charts-pro@v7.23.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-charts@v7.23.0`.

### Tree View

#### `@mui/x-tree-view@v7.23.0`

- [TreeView] React 19 support (#15557) @arminmeh

#### `@mui/x-tree-view-pro@7.23.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-tree-view@7.23.0`.

### Docs

- [docs] Add data caching to lazy loaded detail panel demo (#15555) @cherniavskii

> - [docs] Remove selectors section from list view docs (#15639) @KenanYusuf
> - [docs] Add documentation for the list view feature (#15344) (#15635) @KenanYusuf

Reviewer 4: ```suggestion
- [docs] Add documentation for the list view feature (#15344) @KenanYusuf
```

### Docs

- [docs] Add data caching to lazy loaded detail panel demo (#15555) @cherniavskii
- [docs] Remove selectors section from list view docs (#15639) @KenanYusuf
- [docs][data grid] Add documentation for the list view feature (#15344) (#15635) @KenanYusuf

### Core

- [core] Update @mui/monorepo (#15574) @oliviertassinari

> - React 19 support (#15557) @arminmeh

Reviewer: Last week in the alpha release we added this one to every set of components and removed from `Core`
You can see the diff [here](https://github.com/mui/mui-x/pull/15554/files)

Reviewer 3: I have added the suggestions for all packages
